store/store.go

```diff
@@ -634,8 +648,19 @@ func (s *Store) retryRequest(ctx context.Context, client *http.Client, reqOption
 		if shouldRetryHttpResponse(attempt, resp) {
 			resp.Body.Close()
 			continue
+		} else {
+			err = decode(resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusCreated, resp.Body)
```

Comment:
When we get an error status, and have body with length 0, we may not want to decode as that'll lead to a JSON decode error when we actually want to surface the, say, 500, instead.